utils/rtty/files/rtty.init

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+START=99
+
+BIN=/usr/sbin/rtty
+
+start_rtty() {
+    local cfg="$1"
+    local interface ifname id description host port ssl
+
+    uci_validate_section rtty rtty "${1}" \
+        'interface:uci("network", "@interface"):br-lan' \
+        'id:maxlength(63)' \
+        'description:maxlength(126)' \
+        'host:host' \
+        'port:port' \
+        'ssl:bool:0'
+    
+    [ $? -eq 1 ] && {
+        echo "validation failed" >&2
+        return 1
+    }
+
+    [ -n "$interface" ] && ifname=$(uci -q -p /var/state get network.$interface.ifname)
```

Comment:
`network_get_device` from `/lib/functions/network.sh` is preferred here.  I was told once previously that the uci state mechanism is going to be deprecated.  The usage is as simple as `network_get_device ifname "$interface"`

Reply:
Fixed.